pkg/subvolume/subvolume.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subvolume
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	exec "github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type fsStruct struct {
+	Name string
+	data []string
+}
+
+type subvolume struct {
+	fs     string
+	svg    string
+	subvol string
+}
+
+func List(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace, flag string) {
+
+	pvList, err := clientsets.Kube.CoreV1().PersistentVolumes().List(context.Background(), v1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error fetching PVs: %v\n", err)
+		os.Exit(1)
+	}
+	var subvolumeName []string
+	for _, pv := range pvList.Items {
+		subvolumeName = append(subvolumeName, pv.Spec.CSI.VolumeAttributes["subvolumeName"])
+	}
+
+	sv := getCSIRef(ctx, clientsets, clusterNamespace, operatorNamespace, flag)
+
+	// if stale flag is passed, list only the stale subvolumes
+	if flag == "stale" {
+		// Check whether difference is of the two lists is measurable
+		diff := getDifference(sv, subvolumeName)
+		if diff == nil {
+			logging.Info("No stale volumes found")
+		} else {
+			logging.Info("%s", diff)
+		}
+	} else {
+		logging.Info("%s", sv)
+	}
+}
+
+func getDifference(subvol []subvolume, k8ref []string) []subvolume {
+
+	mb := make(map[string]struct{}, len(k8ref))
+	for _, x := range k8ref {
+		mb[x] = struct{}{}
+	}
+	var diff []subvolume
+	for _, x := range subvol {
+		if _, found := mb[x.subvol]; !found {
+			// reference present in ceph but not in k8s
+			st := subvolume{x.fs, x.subvol, x.svg}
+			diff = append(diff, st)
+		}
+	}
+	if diff == nil {
+		logging.Fatal(fmt.Errorf("No stale Volume found"))
+	}
+
+	return diff
+}
+
+func getCSIRef(ctx context.Context, clientsets *k8sutil.Clientsets, clusterNamespace, operatorNamespace, flag string) []subvolume {
+
+	fsList := exec.RunCommandInOperatorPod(ctx, clientsets, "ceph", []string{"fs", "ls", "--format", "json"}, clusterNamespace, operatorNamespace, true, false)
+	if fsList == "" {
+		logging.Fatal(fmt.Errorf("failed to get filesystem"))
+	}
+
+	fsstruct := unMarshaljson(fsList)
+	var svList string
+	var staleVolume []subvolume
```

Comment:
Looks like this list might contain all subvolumes, depending on the params?
```suggestion
	var subvolumes []subvolume
```